src/nodes.coffee

```diff
@@ -1591,6 +1604,12 @@ exports.Code = class Code extends Base
     @bound       = tag is 'boundfunc'
     @isGenerator = !!@body.contains (node) ->
       (node instanceof Op and node.isYield()) or node instanceof YieldReturn
+    @isAsync     = !!@body.contains (node) =>
+      if (node instanceof Op and node.isAwait()) or node instanceof AwaitReturn
+        node.error "generator function can't contain await" if @isGenerator
```

Comment:
This error check shouldn’t occur inside `@isAsync`. `@isAsync` should just do one thing, and we should check for errors one level up, in the constructor:

```coffee
    @isAsync     = !!@body.contains (node) ->
      (node instanceof Op and node.isAwait()) or node instanceof AwaitReturn

    if @isGenerator and @isAsync
      node.error 'function cannot contain both yield and await'
```
It’s not really that a generator function can’t contain `await`; or that an async function can’t contain `yield`; it’s that no function can contain _both_ `await` and `yield`.